runtime: always use DIP_START and remove duplication of start packages in &rtp (#15632)
neovim/neovim: runtime/doc/options.txt

	   ordering.  This is for preferences to overrule or add to the 
	   distributed defaults or system-wide settings (rarely needed).

							*rtp-packages*
	"start" packages will additionally be used to search for runtime files

> 	after these, but package entries are not visible in `:set rtp`, as the

Reviewer: I guess they also aren't shown by `:echo &rtp` ? Should we go further and say something like:

> :runtime searches "start" packages, thus they are _logically_ but not explicitly part of 'runtimepath'.

> 	final path would be too long and get truncated. use

Reviewer: ```suggestion
	final path would be too long and get truncated. Use
```

	   ordering.  This is for preferences to overrule or add to the 
	   distributed defaults or system-wide settings (rarely needed).

							*rtp-packages*

> 	"start" packages will additionally be used to search for runtime files

Reviewer: wording could be confusing. suggestion:
```suggestion
	"start" packages will also be searched for runtime files after these,
```

> 	"start" packages will additionally be used to search for runtime files
> 	after these, but package entries are not visible in `:set rtp`.

Reviewer: wording could be confusing. suggestion:

```suggestion
	"start" packages will also be searched for runtime files after these,
	though such packages are not explicitly listed in `:set rtp`.
```
